geopandas/io/tests/test_sql.py

```diff
@@ -773,6 +773,20 @@ def test_append_with_different_crs(self, engine_postgis, df_nybb):
         with pytest.raises(ValueError, match="CRS of the target table"):
             write_postgis(df_nybb2, con=engine, name=table, if_exists="append")
 
+    @pytest.mark.parametrize("engine_postgis", POSTGIS_DRIVERS, indirect=True)
```

Comment:
test looks a bit redundant but I did ensure it fails in the previous implentation of using srid = -1 for no CRS cases.

Reply:
Looking at `shapely.get_srid()`, it returns a -1 if it is run on `None`, but 0 when on a geometry with no CRS. The docstring does only specify -1 for 'not-a-geometry value' but does not specify 0 is the no CRS value. Perhaps this misinterpretation was the origin of using -1 as the no CRS value here?

Reply:
Can you include a comment here linking to the issue, so we know why the seemingly redundant test exists?

Reply:
done